### ISSUE TITLE
openmvg develop branch run debug

### ISSUE THREAD

**Issue by sunmengmeng** [openMVG/openMVG#646]
Hi,
I compiled openMVG develop branch successfully.before I run "main_ imagelisting ",
I gave the "focal_pixels " by hand,as follows:
`// double focal_pixels = -1.0;`
`double focal_pixels = 4800.0;`

but when I run the  "main_ imagelisting ",there is a debug error(sorry I don't know why I can't upload the image of the error window):

 Expression: map/set iterator not dereferencable

And then it points to the line 256 in the file "xtree" --->  _DEBUG_ERROR("map/set iterator not dereferencable");

In addition, I tried in the openmvg master ,it is no problem .Is it a debug in the openmvg develop branch ?how can I change the code if I want to give the "focal_pixels " by hand?
Thanks.

**Comment by pmoulon** [comment]
First you can use the -f option to specify the pixel focal length without having to modify the code.
Then, you can use release binaries in order to do not have the Visual Studio compilation error.

Regarding the compilation error in debug, it's certainly because visual studio compiler is checking some rules for some classes when they are used in the STL container. You can identify the problem class and propose a fix.

**Comment by sunmengmeng** [comment]
Thanks very much.I will try it.
In fact , when I use openmvg master and I run the main_globalsfm,I found some poses can't be gotten.for example,I experimented with 185 images ,but I get 146 poses only,though the overlap is large(about 80%).

so, I want to try openmvg develop branches .I hope I can get a more stable result and all poses.This is my next plan,And I will do experiments to find the advantages of the develop branches.

**Comment by pmoulon** [comment]
Sometimes it depends on the input graph.
You can make it denser by using the preset on features to extract more keypoints and so have more matches and more pairs.
See option -p on main_ComputeFeatures.

BTW, develop branch is always better than master, since it contains features that have been validated for the next coming version.

**Comment by sunmengmeng** [comment]
Like I said above,I experimented with 185 images ,but I get 146 poses only.In other words, there are about 30 images not getting poses.
 Then I experimented with the 30 images again,and It's no problem .All poses of the 30 images are computed.
So I think that the first experiment with 185 images maybe have errors in the "globalsfm"step , that leaded to some poses not computed. Is it right?
If my idea is right ,I will research the "globalsfm" step to find the cause of the error.
